src/Week3/Golf.hs

```diff
@@ -0,0 +1,133 @@
+module Week3.Golf where
+
+{-
+ Exercice 1: Hopscotch
+
+ Your first task is to write a function
+    skips :: [a] -> [[a]]
+
+ The output of skips is a list of lists. The first list in the output should
+ be the same as the input list. The second list in the output should
+ contain every second element from the input list. . . and the nth list in
+ the output should contain every nth element from the input list.
+
+ For example:
+  skips "ABCD" == ["ABCD", "BD", "C", "D"]
+  skips "hello!" == ["hello!", "el!", "l!", "l", "o", "!"]
+  skips [1] == [[1]]
+  skips [True,False] == [[True,False], [False]]
+  skips [] == []
+
+ Note that the output should be the same length as the input
+-}
+
+-- We make sure the element always exists by bounding the indices from n-1 to length list - 1
+-- We increment by n each time
+takeNth :: [a] -> Int -> [a]
+-- This doesn't feel much concise. Perhaps there is way to do this
```

Comment:
In addition to not being concise, it's not very readable either. See if you can figure out a way to do this without list comprehension as well. Also note that we try to avoid list comprehension in CDMP.

Reply:
Noted. I've included an implementation that I feel is very straightforward. At least compared to this one. Please have a look at `takeNth'`